lms/static/scripts/frontend_apps/components/EmailPreferencesApp.tsx

```diff
@@ -1,15 +1,41 @@
-import { InfoIcon, Link } from '@hypothesis/frontend-shared';
+import { InfoIcon, Link, ToastMessages } from '@hypothesis/frontend-shared';
 import classnames from 'classnames';
 import { useCallback, useState } from 'preact/hooks';
 
+import type { SelectedDays } from '../config';
 import { useConfig } from '../config';
+import { useToastMessages } from '../hooks/use-toast-messages';
 import EmailPreferences from './EmailPreferences';
 
 export default function EmailPreferencesApp() {
   const { emailPreferences } = useConfig(['emailPreferences']);
-  const [selectedDays, setSelectedDays] = useState(emailPreferences);
+  const [selectedDays, setSelectedDays] = useState(
+    emailPreferences.selectedDays
+  );
   const [saving, setSaving] = useState(false);
   const onSave = useCallback(() => setSaving(true), []);
+  const { toastMessages, dismissToastMessage } = useToastMessages(
+    emailPreferences.result
+      ? [
+          {
+            message: emailPreferences.result,
+            type: 'success',
```

Comment:
So there is no method for the backend to pass an error to the frontend right now. What's the plan if that need does come up? Refactor `emailPreferences.result` to be a `{ message, status }` object?

Reply:
I made the assumption that an error in the BE would end up rendering another template, so we would not need to support that case.

But yeah, making that change would be a simple and reasonable option.